File: snippets/swift/BreezSDKExamples/Sources/Messages.swift
import BreezSDKLiquid

func signMessage(sdk: BindingLiquidSdk) -> SignMessageResponse? {
    // ANCHOR: sign-message
    let signMessageRequest = SignMessageRequest(
        message: "<message to sign>"
    )
    let signMessageResponse = try? sdk
        .signMessage(req: signMessageRequest)

    // Get the wallet info for your pubkey
    let info = try? sdk.getInfo()

    let signature = signMessageResponse!.signature
    let pubkey = info!.walletInfo.pubkey

    print("Pubkey: {}", pubkey);
    print("Signature: {}", signature);
    // ANCHOR_END: sign-message
    return signMessageResponse
}

func checkMessage(sdk: BindingLiquidSdk) -> CheckMessageResponse? {
    // ANCHOR: check-message
    let checkMessageRequest = CheckMessageRequest(
        message: "<message>",
        pubkey: "<pubkey of signer>",
        signature: "<message signature>"
    )
    let checkMessageResponse = try? sdk
        .checkMessage(req: checkMessageRequest)

    let isValid = checkMessageResponse!.isValid

    print("Signature valid: {}", isValid);
    // ANCHOR_END: check-message
    return checkMessageResponse
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/Messages.kt
package com.example.kotlinmpplib

import breez_sdk_liquid.*

class Messages {
    fun signMessage(sdk: BindingLiquidSdk) {
        // ANCHOR: sign-message
        val message = "<message to sign>"
        try {
            val signMessageResponse = sdk.signMessage(SignMessageRequest(message))

            // Get the wallet info for your pubkey
            val info = sdk.getInfo()

            val signature = signMessageResponse?.signature
            val pubkey = info?.walletInfo?.pubkey

            // Log.v("Breez", "Pubkey: ${pubkey}")
            // Log.v("Breez", "Signature: ${signature}")
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: sign-message
    }

    fun checkMessage(sdk: BindingLiquidSdk) {
        // ANCHOR: check-message
        val message = "<message>"
        val pubkey = "<pubkey of signer>"
        val signature = "<message signature>"
        try {
            val checkMessageRequest = CheckMessageRequest(message, pubkey, signature)
            val checkMessageResponse = sdk.checkMessage(checkMessageRequest)

            val isValid = checkMessageResponse?.isValid

            // Log.v("Breez", "Signature valid: ${isValid}")
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: check-message
    }

}
